### PR TITLE
[FEATURE] Update ephemeral pandas datasource to return a Batch instead of a Validator.

### PR DESCRIPTION
**Pull request by billdirks**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/datasource/fluent/pandas_datasource.py [hunk 1]
```diff
@@ -671,8 +670,8 @@ def _get_validator(
         else:
             batch_request = asset.build_batch_request()
 
-        # TODO: raise error if `_data_context` not set
-        return self._data_context.get_validator(batch_request=batch_request)  # type: ignore[union-attr] # self._data_context must be set
+        # BDIRKS, should only grab the last one, maybe do something better than this implementation
```
Comment (billdirks) [inline great_expectations/datasource/fluent/pandas_datasource.py]: Delete note to self

File: tests/checkpoint/test_checkpoint.py [hunk 2]
```diff
@@ -4869,40 +4868,6 @@ def test_context_checkpoint_crud_conflicting_validator_and_validation_args_raise
     assert "either a validator or validations list" in str(e.value)
 
 
-@pytest.mark.filesystem
```
Comment (billdirks) [inline tests/checkpoint/test_checkpoint.py:4872]: We want to stop passing a validator to a checkpoint so I'm removing this test.

File: tests/integration/docusaurus/tutorials/quickstart/quickstart.py [hunk 3]
```diff
@@ -10,33 +14,34 @@
...
 # </snippet>
 
 # Create Expectations
 # <snippet name="tutorials/quickstart/quickstart.py create_expectation">
-validator.expect_column_values_to_not_be_null("pickup_datetime")
-validator.expect_column_values_to_be_between(
-    "passenger_count", min_value=1, max_value=6
+suite = context.add_expectation_suite("my_suite")
+
+# TODO: update where these expectations are imported
```
Comment (billdirks) [inline tests/integration/docusaurus/tutorials/quickstart/quickstart.py:26]: I'm tracking removing these TODOs in https://greatexpectations.atlassian.net/browse/V1-111

File: great_expectations/datasource/fluent/pandas_datasource.py [hunk 4]
```diff
@@ -657,9 +656,9 @@ def _validate_asset_name(asset_name: Optional[str] = None) -> str:
             asset_name = DEFAULT_PANDAS_DATA_ASSET_NAME
         return asset_name
 
-    def _get_validator(
+    def _get_batch(
```
Comment (billdirks) [inline great_expectations/datasource/fluent/pandas_datasource.py:660]: All the functional changes are in this file. The other changes are tests/docs/pyi related.

File: tests/datasource/fluent/test_pandas_datasource.py [hunk 5]
```diff
@@ -357,8 +357,15 @@ def test_positional_arguments(
...
+        # We don't care about the actual data being read in and the batch that will be produced from that data.
+        # In fact, we call all our read methods on a path which might not be readable by the reader (eg calling
+        # read_json on a csv file). We patch the internal call that actually tries to read and create the batch.
+        # Ideally, we would rewrite this test so we wouldn't need to mock like this.
         mocker.patch(
-            "great_expectations.data_context.data_context.abstract_data_context.AbstractDataContext.get_validator"
+            "great_expectations.datasource.fluent.pandas_datasource._PandasDataAsset.get_batch_list_from_batch_request"
         )
         _ = read_method(*positional_args.values())
         # read_* returns a validator, but we just want to inspect the asset
```
Comment (joshua-stauffer) [inline tests/datasource/fluent/test_pandas_datasource.py]:
```suggestion
        # read_* returns a batch, but we just want to inspect the asset
```

Comment (joshua-stauffer) [reply tests/datasource/fluent/test_pandas_datasource.py]: probably don't need to retrieve the asset one line below this, could just use the asset returned by the batch

File: tests/datasource/fluent/test_viral_snippets.py [hunk 6]
```diff
@@ -300,27 +304,28 @@ def test_quickstart_workflow(
...
+    result = batch.validate(suite)
+
+    assert result.success
 
+    # TODO: Add mechanism to view results.
     # View results
-    mock_open = mocker.patch("webbrowser.open")
-    context.view_validation_result(checkpoint_result)
+    # mock_open = mocker.patch("webbrowser.open")
+    # context.view_validation_result(checkpoint_result)
```
Comment (joshua-stauffer) [inline tests/datasource/fluent/test_viral_snippets.py]: do you intend to leave these in the PR?

**Review by joshua-stauffer — CHANGES_REQUESTED** [review]
thanks @billdirks, in general looks good! left a couple small comments. I wasn't able to comment on it because its on unchanged lines, but the docstrings for all these pandas methods need to be updated.

**Review by joshua-stauffer — APPROVED** [review]
thanks for addressing comments!

**Pull request merged**